app/views/handlers/question.py

```diff
@@ -194,7 +226,27 @@ def get_list_summary_context(self):
     def handle_post(self):
         # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
-        self.questionnaire_store_updater.update_answers(self.form.data)
+        dynamic_answers = False
+        for item in self.form.data:
+            if list_item_id := item.split("-")[-1]:
+                if (
+                    list_item_id
+                    in self.questionnaire_store_updater._list_store._list_item_ids()
+                ):
+                    item_id = item.replace(f"-{list_item_id}", "")
+                    form_data = {item_id: self.form.data[item]}
+                    self.questionnaire_store_updater.update_answers(
+                        form_data, list_item_id
+                    )
+                    dynamic_answers = True
+                elif item in self._schema.get_answer_ids_for_question(
+                    self.rendered_block["question"]
+                ):
+                    form_data = {item: self.form.data[item]}
+                    self.questionnaire_store_updater.update_answers(form_data)
+
+        if not dynamic_answers:
+            self.questionnaire_store_updater.update_answers(self.form.data)
```

Comment:
Regarding your last sentence: how do we know answer in form is dynamic? Do we look it up in dynamic answers key in in that resolved block? Check if we get a match?

Reply:
I don't do this in `update_answers` cause it updates for each list item id and here in this `form.data` we (can) have several.

Reply:
We still shouldn't be mutating the form. Runner should be able to work out if it is dynamic or not and handle it accordingly.

This can be reverted and `update_answers` can be updated to handle this in a much simpler way due to the changes we made to how placeholder handles this.

```py
def update_answers(
    self, form_data: Mapping[str, Any], list_item_id: Optional[str] = None
) -> None:
    list_item_id = list_item_id or self._current_location.list_item_id
    answers_by_answer_id = self._schema.get_answers_for_question_by_id(self._current_question)

    for answer_id, answer_value in form_data.items():
        if answer_id not in answers_by_answer_id:
            continue

        resolved_answer = answers_by_answer_id[answer_id]
        answer_id_to_use = resolved_answer.get("original_answer_id") or answer_id
        list_item_id_to_use = resolved_answer.get("list_item_id") or list_item_id

        answer_updated = self._update_answer(answer_id_to_use, list_item_id_to_use, answer_value)
        if answer_updated:
            self._capture_section_dependencies_for_answer(answer_id_to_use)
            self._capture_block_dependencies_for_answer(answer_id_to_use)
```

With this change you can remove the form manipulation and let update answer handles answer updation.